### PR TITLE
refactor: remove buffer pool manager instance

### PR DESCRIPTION
**Pull request by skyzh**
BufferPoolManagerInstance is originally used by project 3 when checking if NLJ is implemented correctly by repeatively scanning all the pages. Given now we have a more powerful execution framework we can actually check if NLJ's child is initialized for multiple times, or check how many times the disk pages are read.

Signed-off-by: Alex Chi <iskyzh@gmail.com>

DO NOT PUSH PROJECT SOLUTIONS PUBLICLY.

And especially do NOT open pull requests with project solutions!

Please be considerate of this free educational resource.

### CODE REVIEW

File: test/catalog/catalog_test.cpp [hunk 1]
```diff
@@ -14,7 +14,7 @@
 #include <unordered_set>
 #include <vector>
 
-#include "buffer/buffer_pool_manager_instance.h"
```
Comment (yliang412) [inline test/catalog/catalog_test.cpp:17]: I think some of the refactor is not synced?

Comment (yliang412) [reply test/catalog/catalog_test.cpp:17]: Some of the `BufferPoolManagerInstance` stuff is still present.

Comment (skyzh) [reply test/catalog/catalog_test.cpp:17]: Fixing now...

**Review by yliang412 — APPROVED** [review]
LGTM.

**Comment by LzwY25** [comment]
When I submit project1 to gradescope. it requires buffer_pool_manager_instance.h and buffer_pool_manager_instance.cpp, but I can't find these files.
How can I submit project1 after removing buffer pool manager instance?

**Comment by skyzh** [comment]
Try using an earlier commit. Generally we won't be maintaining projects from previous semesters when a new semester begins.

**Pull request merged**